src/pusher.jl

```diff
@@ -0,0 +1,90 @@
+# Native pusher
+
+struct TraceProblem{T<:Real, TP}
+   stateinit::AbstractVector{T}
+   tspan::Tuple{T, T}
+   param::TP
+end
+
+"updates velocity using the Boris method, Birdsall, Plasma Physics via Computer Simulation, p.62"
```

Comment:
Should we cite the original paper of Boris?
https://apps.dtic.mil/sti/citations/ADA023511

Reply:
Yes we should! I will clean up the doc later.